plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/DiskCache.kt

```diff
@@ -203,11 +267,20 @@ class DiskCache(
 
     private fun writeKey(path: Path, consumer: (OutputStream) -> Unit) {
         LOG.debug { "writing to $path" }
-        path.tryDirOp(LOG) { createParentDirectories() }
+        try {
+            path.tryDirOp(LOG) { createParentDirectories() }
 
-        path.tryFileOp(LOG) {
-            touch(restrictToOwner = true)
-            outputStream().use(consumer)
+            path.tryFileOp(LOG) {
+                touch(restrictToOwner = true)
+                outputStream().use(consumer)
+            }
+        } catch (e: Exception) {
+            AwsTelemetry.saveCredentials(
```

Comment:
Did we agree to use save credentials here or in the parent function? Was it save/modify credentials? @dhruvigajjar